Kitodo/src/main/java/org/kitodo/production/forms/CommentForm.java

```diff
@@ -0,0 +1,281 @@
+     */
+    public void setCorrectionTaskId(String correctionTaskId) {
+        this.correctionTaskId = correctionTaskId;
+    }
+
+    /**
+     * Correction message to previous Tasks.
+     */
+    public List<Task> getPreviousStepsForProblemReporting() {
+        refreshProcess(this.process);
+        return ServiceManager.getTaskService().getPreviousTasksForProblemReporting(
+                ServiceManager.getProcessService().getCurrentTask(this.process).getOrdering(), this.process.getId());
+    }
+
+    public int getSizeOfPreviousStepsForProblemReporting() {
+        return getPreviousStepsForProblemReporting().size();
+    }
+
+    /**
+     * Solve the problem.
+     */
+    public void solveProblem(Comment comment) {
+        try {
+            ServiceManager.getWorkflowControllerService().solveProblem(comment);
+        } catch (DataException e) {
+            Helper.setErrorMessage("SolveProblem", logger, e);
+        }
+        refreshProcess(this.process);
+    }
+
+    /**
+     * Solve the problem to all batch processes.
+     */
+    public void solveProblemForAll(Comment comment) {
+        for (Task task : batchHelper.getSteps()) {
+            for (Comment processComment : ServiceManager.getCommentService().getAllCommentsByProcess(task.getProcess())) {
+                if (!processComment.isCorrected()
+                        && processComment.getCorrectionTask().getTitle().equals(comment.getCorrectionTask().getTitle())) {
+                    solveProblem(processComment);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get process.
+     *
+     * @return value of process
+     */
+    public Process getProcess() {
+        return process;
+    }
+
+    /**
+     * Set process.
+     *
+     * @param process as org.kitodo.data.database.beans.Process
+     */
+    public void setProcess(Process process) {
+        this.process = process;
+    }
+
+    /**
+     * refresh the process in the session.
+     *
+     * @param process Object process to refresh
+     */
+    public void refreshProcess(Process process) {
+        try {
+            if (!Objects.equals(process.getId(), 0)) {
+                ServiceManager.getProcessService().refresh(process);
+                setProcess(ServiceManager.getProcessService().getById(process.getId()));
+            }
+
+        } catch (DAOException e) {
+            Helper.setErrorMessage("Unable to find process with ID " + process.getId(), logger, e);
+        }
+    }
+
+
+    /**
+     * Get processId.
+     *
+     * @return value of processId
+     */
+    public String getProcessId() {
+        return processId;
+    }
+
+    /**
+     * Set processId.
+     *
+     * @param processId as java.lang.String
+     */
+    public void setProcessId(String processId) {
+        this.processId = processId;
+        try {
+            setProcess(ServiceManager.getProcessService().getById(Integer.parseInt(this.processId)));
```

Comment:
As a setter is called very often from Frontend, i would rather get the process by id on the points, when you really need it.
For example on line 49, in getAllCommentsByProcess you only need the ID in the end, you don't need to get the process from database beforehand.